src/build_system.ml

```diff
@@ -1199,8 +1204,11 @@ let all_lib_deps t ~request =
 
 let all_lib_deps_by_context t ~request =
   let targets = static_deps_of_request t request in
-  List.fold_left (rules_for_targets t targets) ~init:[] ~f:(fun acc rule ->
-    let lib_deps = Build_interpret.lib_deps rule.Internal_rule.build in
+  List.fold_left (rules_for_targets t targets) ~init:[] ~f:(fun acc (rule : Internal_rule.t) ->
+      let lib_deps =
+        match Build_interpret.lib_deps rule.build with
+        | None -> Pmap.empty
+        | Some deps -> Pmap.singleton rule.dir deps in
```

Comment:
I thought about maybe cleaning up this code by adding a `Map.add_opt` function. But I see that no such thing is in base/core, so I'm not sure how much sense that makes.

Reply:
I think the current code is OK